File: Examples/Node.js/generating-saving/without-cloud-storage/create-barcode-and-get-image-as-stream.js
var fs = require('fs');
var assert = require('assert');
var BarcodeApi = require('asposebarcodecloud');
var StorageApi = require('asposestoragecloud');

var configProps = require('../../Config/config.json');
var data_path = '../../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;

// ExStart:1
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose.Storage API SDK
var storageApi = new StorageApi(config);

// Instantiate Aspose.BarCode API SDK
var barcodeApi = new BarcodeApi(config);

// Set Filename of image
var name = "sample-barcode.png";

// Set Text to encode inside barcode
var text = "Aspose.BarCode for Cloud";

// Set Barcode Symbology
var type = "datamatrix";

// Set Barcode Image Format
var format = "PNG";

try {
	// Upload source file to aspose cloud storage
	storageApi.PutCreate(name  , null, null, data_path + name  , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.BarCode Cloud SDK API to create barcode and save image to a stream        
	barcodeApi.GetBarcodeGenerate(text, type, format, null, null, null, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			// Download generated barcode from api response
			var outfilename = name + "." + format;
			var writeStream = fs.createWriteStream(outFolder + outfilename);
			writeStream.write(responseMessage.body);
				
	});
});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Perl/generating-saving/without-cloud-storage/create-barcode-and-get-image-as-stream.pl
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposeBarCodeCloud::BarcodeApi;
use AsposeBarCodeCloud::ApiClient;
use AsposeBarCodeCloud::Configuration;

    
my $configFile = '../../config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);

my $data_path = '../../../../Data/';
my $out_path = $configProps->{'out_folder'};

#ExStart:1
$AsposeBarCodeCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeBarCodeCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeBarCodeCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};


# Instantiate Aspose.Storage API SDK 
my $storageApi = AsposeStorageCloud::StorageApi->new();

# Instantiate Aspose.BarCode API SDK
my $barcodeApi = AsposeBarCodeCloud::BarcodeApi->new();

# Set Filename of image
my $name = 'sample-barcode.png';

# Set Text to encode inside barcode
my $text = 'Aspose.BarCode for Cloud';

# Set Barcode Symbology
my $type = 'datamatrix';

# Set Barcode Image Format
my $format = 'png';

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);

# Invoke Aspose.BarCode Cloud SDK API to create barcode and save image to a stream           
$response = $barcodeApi->GetBarcodeGenerate(text => $text, type => $type, format => $format);

if($response->{'Status'} eq 'OK'){
	# Download generated barcode from api response 
	my $output_file = $out_path . $name . '.' . $format;	
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1